src/support/rand.c

```diff
@@ -68,7 +68,7 @@ __wt_random_init(WT_RAND_STATE volatile *rnd_state) WT_GCC_FUNC_ATTRIBUTE((visib
     M_W(rnd) = DEFAULT_SEED_W;
     M_Z(rnd) = DEFAULT_SEED_Z;
 
-    *rnd_state = rnd;
+    WT_WRITE_ONCE(*rnd_state, rnd);
```

Comment:
I'm confused about why this helps. Do you think the results of line 68 (`M_W`) could be written back to `*rnd_state` before the `M_Z` assignment? It feels like if we need to protect against that, it would need something different (i.e: protection on lines 68 and 69). This has also picked a bit of code that already uses a volatile - is this applying volatile to the local `rnd` as well? If so, I think the code would be clearer just declaring it as such with a big comment, rather than hiding it in a macro.

Reply:
From prior conversation with Luke this was inspired by a change in WT-11414 since [this `ORDERED_READ`](https://github.com/wiredtiger/wiredtiger/pull/9701/files#diff-3bd3eece4c573df307d47893bf126eb7d557abd623250c70d72e0653da0baa0eR148) is becoming a `READ_ONCE`and it made sense to for the writer side to also change. The case it protected against was 
```
M_W(rnd) = DEFAULT_SEED_W;
*rnd_state = rnd;
M_Z(rnd) = DEFAULT_SEED_Z;
*rnd_state = rnd;
```

I didn't notice that `rnd_state` was a volatile pointer though.. `WRITE_ONCE` only casts the destination variable so it makes this change redundant. This also raises questions why we have the `WT_ORDERED_READ` in the first place so I'll need to dig into this.